### PR TITLE
Add support for pgadmin container.

### PR DESCRIPTION
**Pull request by avrittrohwer**
Terraform implementation for https://github.com/civiform/civiform/issues/2563.

New resources are disabled by default by using count = 0 on the module.

This will be followed up with an additional PR that sets count = 1 via a python command wrapper.

Tested:
1. Set count to 1, built a new docker image, ran `USE_DOCKER=true bin/deploy --tag=latest` in my config repo.
2. Terraform successfully deployed the pgadmin resources.
3. I navigated to https://avritt-dev.civiform.dev:4433 and successfully logged in using the created username and password.
4. I opened the CiviForm db group which prompted for the database password.  I successfully authenticated to the postgres database.
5. I successfully queried the postgres database with `SELECT * FROM questions` which returned the 2 default questions.
6. Set the count to 0, built a new docker image, ran `USE_DOCKER=true bin/deploy --tag=latest` in my config repo.
7. Terraform successfully destroyed the pgadmin resources.
8. I manually checked the lb listener rules, target groups, ecs services, ecs tasks, and secrets in the AWS UI to ensure the resources were properly deleted.

### CODE REVIEW

File: cloud/aws/modules/pgadmin/main.tf [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    from_port       = 5432
+    to_port         = 5432
+    protocol        = "tcp"
+    security_groups = [var.db_sg_id]
+  }
+}
+
+# Run a pgadmin container via a ecs service.
+resource "aws_ecs_service" "pgadmin" {
+  name            = "${var.app_prefix}-pgadmin"
```
Comment (nbeloglazov) [inline cloud/aws/modules/pgadmin/main.tf:77]: Can we use local.name_prefix here as well for consistency? Or you intentionally didn't use "civiform" prefix here?

File: cloud/aws/modules/pgadmin/task.tf [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  source    = "cn-terraform/cloudwatch-logs/aws"
+  version   = "1.0.12"
+  logs_path = "${var.app_prefix}-pgadmin-logs/"
+  tags = {
+    Name = "${var.app_prefix} pgAdmin Cloud Watch Logs"
+    Type = "pgAdmin Cloud Watch Logs"
+  }
+}
+
+module "pgadmin_init_container_def" {
```
Comment (nbeloglazov) [inline cloud/aws/modules/pgadmin/task.tf:91]: Can you add comment explaining the purpose of this container? Given that it's pretty odd container (just used to generate json file).

File: cloud/aws/modules/pgadmin/secrets.tf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  secret_id     = aws_secretsmanager_secret.pgadmin_username_secret.id
+  secret_string = "${random_password.pgadmin_username.result}@default.login"
+}
+
+# Create a random generated password to use for pgadmin password.
+resource "random_password" "pgadmin_password" {
+  length  = 40
+  special = false
+  keepers = {
+    version = 1
```
Comment (nbeloglazov) [inline cloud/aws/modules/pgadmin/secrets.tf:28]: What does keepers { version = 1 } do?

**Review by bion — CHANGES_REQUESTED** [review]

File: cloud/aws/modules/pgadmin/main.tf [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    description     = "postgres egress"
+    from_port       = 5432
+    to_port         = 5432
+    protocol        = "tcp"
+    security_groups = [var.db_sg_id]
+  }
+}
+
+# Run a pgadmin container via a ecs service.
+resource "aws_ecs_service" "pgadmin" {
```
[the remark this answers is not in the document]
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/main.tf:76]: It looks like the [terraform resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service#task_definition) only supports one task definition per service, so if we wanted to run another container we would have to add another service for it.

File: cloud/aws/modules/pgadmin/main.tf (hunk 1 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/main.tf]: Done.

File: cloud/aws/modules/pgadmin/task.tf (hunk 2 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/task.tf:96]: Done.

File: cloud/aws/modules/pgadmin/secrets.tf (hunk 3 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/secrets.tf:28]: It is an arbitrary map of values that forces recreation when changed: https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/password#keepers

I copied these secrets from the existing pattern: https://github.com/civiform/cloud-deploy-infra/blob/main/cloud/aws/templates/aws_oidc/secrets.tf#L27

File: cloud/aws/modules/pgadmin/main.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (bion) [reply cloud/aws/modules/pgadmin/main.tf:76]: Ah good to know, thanks

File: cloud/aws/modules/pgadmin/secrets.tf (hunk 3 above)
Comment (nbeloglazov) [reply cloud/aws/modules/pgadmin/secrets.tf:28]: Is it possible/would it makes sense to re-generate new password on each run? That way if password is leaked like admin shares it accidentally - it won't be useful next time pgadmin is deployed.

File: cloud/aws/modules/pgadmin/task.tf (hunk 2 above)
Comment (nbeloglazov) [reply cloud/aws/modules/pgadmin/task.tf:96]: Another question related to that. Is there a race condition between the first container (that writes server.json) and the second (pgadmin)? Meaning that first container must run before the second initializes?

Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/task.tf:96]: There would be a race condition if I did not configure:

```
  container_depends_on = [
    { containerName = "${local.name_prefix}-init", condition = "COMPLETE" }
  ]
```

in the pgadmin_container_def

Comment (nbeloglazov) [reply cloud/aws/modules/pgadmin/task.tf:96]: Ah, cool!

File: cloud/aws/modules/pgadmin/secrets.tf (hunk 3 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/secrets.tf:28]: The current behavior is that if the pgadmin secrets have been deployed then destroyed, the next time they get deployed a new random password will be generated

Comment (nbeloglazov) [reply cloud/aws/modules/pgadmin/secrets.tf:28]: Ah. Perfect!

**Review by nbeloglazov — APPROVED** [review]

File: cloud/aws/modules/pgadmin/task.tf [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    module.pgadmin_init_container_def.json_map_object,
+    module.pgadmin_container_def.json_map_object
+  ])
+
+  execution_role_arn       = aws_iam_role.civiform_pgadmin_task_execution_role.arn
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+
+  volume {
+    name = "config"
```
Comment (avrittrohwer) [inline cloud/aws/modules/pgadmin/task.tf:17]: Put volume name in a terraform local

File: cloud/aws/modules/pgadmin/main.tf [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  protocol          = "tcp"
+}
+
+# Add new listener to existing load balancer for the main civiform server.
+#
+# We do this to reuse the exiting TLS infra so that we do not need to manage a separate certificate.
+resource "aws_lb_listener" "pgadmin" {
+  load_balancer_arn = var.lb_arn
+  port              = 4433
+  protocol          = "HTTPS"
```
Comment (avrittrohwer) [inline cloud/aws/modules/pgadmin/main.tf:18]: Double check that a http request fails

File: cloud/aws/modules/pgadmin/main.tf [hunk 7]
```diff
@@ -0,0 +1,165 @@
+# Allow load balancer to accept traffic on new port.
+resource "aws_security_group_rule" "pgadmin_lb" {
+  description       = "Allow tcp traffic on port 4433"
+  security_group_id = var.lb_access_sg_id
+  cidr_blocks       = ["0.0.0.0/0"] # Allow traffic from all IPs.
```
Comment (avrittrohwer) [inline cloud/aws/modules/pgadmin/main.tf:5]: Make this a variable configurable via the config.sh file

**Review by bion — CHANGES_REQUESTED** [review]
requested changes in gvc session

File: cloud/aws/modules/pgadmin/main.tf (hunk 6 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/main.tf:18]: HTTP traffic fails:

```
$ curl http://avritt-dev.civiform.dev:4433
<html>
<head><title>400 The plain HTTP request was sent to HTTPS port</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<center>The plain HTTP request was sent to HTTPS port</center>
</body>
</html>
```

File: cloud/aws/pgadmin/build-pgdmin [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/bash
+
+# Builds the customized pgadmin Docker image.
+
+docker build \
+  -t docker.io/avritt/pgadmin:avritt-dev \
+  -f pgadmin.Dockerfile .
+
```
Comment (avrittrohwer) [inline cloud/aws/pgadmin/build-pgdmin:8]: Replace avritt with civiform, add a github action to build new images with latest tag

Comment (avrittrohwer) [reply cloud/aws/pgadmin/build-pgdmin:8]: Done.

File: cloud/aws/modules/pgadmin/task.tf (hunk 5 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/task.tf]: Done.

File: cloud/aws/modules/pgadmin/main.tf (hunk 7 above)
Comment (avrittrohwer) [reply cloud/aws/modules/pgadmin/main.tf]: Done.

**Review by bion — APPROVED** [review]
This looks great!

**Pull request merged**